src/components/scoring/XPSystem.tsx

```diff
@@ -0,0 +1,24 @@
+import { useEffect } from "react";
+import { useUser } from "@src/state/useUser";
+
+const XPSystem = () => {
+  const userHP = useUser((state) => state.user.hp);
+  const currentPosture = useUser((state) => state.currentPosture);
+  const setXP = useUser((state) => state.setXP);
+
+  useEffect(() => {
+    const interval = setInterval(() => {
```

Comment:
you might want to check if the tracking is enabled to activate the interval, and also clear the interval when the real-time tracking is disabled.

another way could be to check if the postureData length is multiple of 5, so every 5 readings you check the currentPosture, the postureData list will no be updated if the isTrackingEnabled flag is false.

Reply:
Thank you for this, I like the second approach more actually! Implemented here: https://github.com/LangaraBeetles/weaup-react-native-app/pull/31/commits/2f7d0facf9c959a4e9753fc9365c44f4d5a3dc65